src/main/java/woowacourse/shoppingcart/application/OrderService.java

```diff
@@ -23,64 +23,71 @@ public class OrderService {
     private final OrderDao orderDao;
     private final OrdersDetailDao ordersDetailDao;
     private final CartItemDao cartItemDao;
-    private final CustomerDao customerDao;
     private final ProductDao productDao;
 
     public OrderService(final OrderDao orderDao, final OrdersDetailDao ordersDetailDao,
-                        final CartItemDao cartItemDao, final CustomerDao customerDao, final ProductDao productDao) {
+                        final CartItemDao cartItemDao, final ProductDao productDao) {
         this.orderDao = orderDao;
         this.ordersDetailDao = ordersDetailDao;
         this.cartItemDao = cartItemDao;
-        this.customerDao = customerDao;
         this.productDao = productDao;
     }
 
-    public Long addOrder(final List<OrderRequest> orderDetailRequests, final String customerName) {
-        final Long customerId = customerDao.findIdByUserName(customerName);
-        final Long ordersId = orderDao.addOrders(customerId);
+    public Long save(final List<OrderRequest> orderDetailRequests, final Long customerId) {
+        final Long orderId = orderDao.save(customerId);
+        List<Long> cartItemIds = collectCartItemIds(orderDetailRequests);
+        List<CartItem> cartItems = cartItemDao.findByIdsIn(cartItemIds);
+        List<OrderDetail> orderDetails = collectOrderDetails(cartItems);
 
-        for (final OrderRequest orderDetail : orderDetailRequests) {
-            final Long cartId = orderDetail.getCartId();
-            final Long productId = cartItemDao.findProductIdById(cartId);
-            final int quantity = orderDetail.getQuantity();
+        reduceOrderedStock(orderDetails, findProductsInCartItems(cartItems));
 
-            ordersDetailDao.addOrdersDetail(ordersId, productId, quantity);
-            cartItemDao.deleteCartItem(cartId);
-        }
+        ordersDetailDao.saveAll(orderId, orderDetails);
+        cartItemDao.deleteByIdsIn(cartItemIds, customerId);
 
-        return ordersId;
+        return orderId;
     }
 
-    public Orders findOrderById(final String customerName, final Long orderId) {
-        validateOrderIdByCustomerName(customerName, orderId);
-        return findOrderResponseDtoByOrderId(orderId);
+    private List<OrderDetail> collectOrderDetails(List<CartItem> cartItems) {
+        return cartItems.stream().map(OrderDetail::from).collect(Collectors.toList());
     }
 
-    private void validateOrderIdByCustomerName(final String customerName, final Long orderId) {
-        final Long customerId = customerDao.findIdByUserName(customerName);
+    private List<Long> collectCartItemIds(List<OrderRequest> orderDetailRequests) {
+        return orderDetailRequests.stream().map(OrderRequest::getCartItemId)
+                .collect(Collectors.toList());
+    }
```

Comment:
```suggestion
    private List<Long> collectCartItemIds(List<OrderRequest> orderDetailRequests) {
        return orderDetailRequests.stream()
                .map(OrderRequest::getCartItemId)
                .collect(Collectors.toList());    }
```

적절한 줄바꿈은 가독성에 아주 많은 영향을 줘요!